src/main/java/edu/stanford/protege/webprotege/icd/RetiredClassCheckerImpl.java

```diff
@@ -0,0 +1,54 @@
+package edu.stanford.protege.webprotege.icd;
+
+import edu.stanford.protege.webprotege.DataFactory;
+import edu.stanford.protege.webprotege.index.AnnotationAssertionAxiomsIndex;
+import org.jetbrains.annotations.NotNull;
+import org.semanticweb.owlapi.model.*;
+
+import javax.inject.Inject;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class RetiredClassCheckerImpl implements RetiredClassChecker {
+
+    private final AnnotationAssertionAxiomsIndex annotationAssertionAxiomsIndex;
+
+
+    @Inject
+    public RetiredClassCheckerImpl(AnnotationAssertionAxiomsIndex annotationAssertionAxiomsIndex) {
+        this.annotationAssertionAxiomsIndex = checkNotNull(annotationAssertionAxiomsIndex);
+    }
+
+    @Override
+    public boolean isRetired(@NotNull OWLEntity entity) {
+
+        String titleValue = getIcdTitleValueForEntity(entity);
+
+        return titleValue.contains(IcdConstants.RETIRED_TITLE_PART);
+    }
+
+    private String getIcdTitleValueForEntity(OWLEntity entity) {
+
+        OWLLiteral titleLiteral = getOwlLiteralWithTitleValueFromEntity(entity);
+
+        var titleValue = titleLiteral.getLiteral();
+
+        return titleValue;
+    }
+
+    private OWLLiteral getOwlLiteralWithTitleValueFromEntity(OWLEntity entity) {
+        var titleAndLabelAnnotations = annotationAssertionAxiomsIndex.getAnnotationAssertionAxioms(entity.getIRI())
```

Comment:
[Question] so we didn't make a generic way of filtering on multiple layers? maybe a technical dept can be created out of this.